docs/developer/release-process.md

```diff
@@ -55,14 +55,9 @@ To create a new release, follow these steps:
       created. If included, use the Release Notes specified in a PR.
       - If the supported Gateway API minor version has changed since the last release, add a note to the release notes explaining if the previous version is no longer supported.
       - Merge the release PR once it has received all necessary approvals.
-6. Ensure you are on the latest version of the release branch and are up-to-date on all commits, then create and push the release tag in the format `vX.Y.Z`.
-
-   ```shell
-   git tag vX.Y.Z
-   git push upstream vX.Y.Z
-   ```
-
+6. Run the [Production Release](https://github.com/nginx/nginx-gateway-fabric/actions/workflows/production-release.yml) workflow with the correct tag e.g. `v2.1.0`
```

Comment:
To prevent any potential premature releases, I would preface this with something like "Once you are ready to officially publish the release..."